libs/zedUpload/awsutil/s3sync.go

```diff
@@ -220,19 +231,30 @@ func getNeededParts(cWriterOptions *writerOptions, bname, bkey string, doneParts
 				partInd:             i,
 				offset:              S3PartSize*i + currentPartSize,
 				writtenBytes:        currentPartSize,
+				maxPartSize:         S3PartSize,
 			},
 			bname: bname,
 			bkey:  bkey,
 			start: S3PartSize*i + currentPartSize,
 			size:  S3PartSize - currentPartSize,
 		}
 		if i == partsCount-1 {
+			part.cWriter.maxPartSize = size - (partsCount-1)*S3PartSize
```

Comment:
Why do you need to update maxPartSize? Shouldn't it be fixed? Or is this the last part which needs a different size?

Reply:
Correct, the last part has different max size in general.

Reply:
Would be good to add a comment to that affect e.g., // Set maxPartSize for last part to the remaining size

Reply:
Modified